lib/game_validator.rb

```diff
@@ -0,0 +1,78 @@
+require 'yaml'
+
+class GameValidator
+  def initialize(display, game_loader)
+    @display = display
+    @game_loader = game_loader
+  end
+
+  def validate_game_type_input(filename)
+    if existing_games_empty?(filename)
+      'new'
+    else
+      retrieve_game_type
+    end
+  end
+
+  def new_game_name(filename)
+    game_name = prompt_for_name
+    until game_name_not_exists?(filename, game_name)
+      game_name = prompt_for_new_name
+    end
+    game_name
+  end
+
+  def existing_game_selection(filename)
+    existing_games = @game_loader.existing_game_names(filename)
+    @display.existing_game_names(existing_games)
+
+    game_name = prompt_for_name
+    until game_name_exists?(filename, game_name)
+      game_name = prompt_for_correct_name
+    end
+    game_name
+  end
+
+  private
+
+  def retrieve_game_type
```

Comment:
why not `get_game_type`? (non-native speaker here :P)

Reply:
I agree. `get_game_type` is clearer than `retrieve_game_type`. I'll change this, thanks